pkg/process/checks/container.go

```diff
@@ -59,8 +60,12 @@ func (c *ContainerCheck) Init(_ *SysProbeConfig, info *HostInfo) error {
 
 // IsEnabled returns true if the check is enabled by configuration
 func (c *ContainerCheck) IsEnabled() bool {
-	// TODO - move config check logic here
-	return true
+	// The process and container checks are mutually exclusive
+	if ddconfig.Datadog.GetBool("process_config.process_collection.enabled") || !ddconfig.IsAnyContainerFeaturePresent() {
+		return false
+	}
+
```

Comment:
Since we're removing `pkg/process/runner/checks.go`, we should migrate this warning log to the container check as well: `log.Warn("Disabled container check because no container environment detected (see list of detected features in `agent status`)")`

Reply:
Oh great point! I added it to the Container check, but not the ContainerRT check so we wouldn't get a duplicated message